### PR TITLE
Correct return type for `std` along region

### PR DESCRIPTION
**Pull request by omus**
Addresses part of #25300

### CODE REVIEW

**Review by andreasnoack — APPROVED** [review]

**Comment by nalimilan** [comment]
Why should the return type of `std` differ from that of `var`?

**Comment by andreasnoack** [comment]
> Why should the return type of `std` differ from that of `var`?

Because `std` has a `sqrt` and `var` hasn't.

**Comment by nalimilan** [comment]
I figured that, but that doesn't sound so obvious to me. Anyway I guess it makes sense to return whatever type `sqrt` returns.

File: test/statistics.jl [hunk 1]
```diff
@@ -159,6 +159,12 @@ end
         @test varm(A, 0.) ≈ sum(map(abs2, A)) / (length(A) - 1)
         @test varm(A, mean(A)) ≈ var(A)
     end
+
+    @test typeof(var([1//1, 2//1])) == Rational{Int}
```
Comment (TotalVerb) [inline test/statistics.jl]: `isa` rather than `typeof(...) == ...`

**Pull request merged**